### PR TITLE
ath79: add support for D-Link DIR-859 A3

### PR DESCRIPTION
**Pull request by DragonBluep**
```
Specifications:
  SOC:      QCA9563 775 MHz + QCA9880
  Switch:   QCA8337N-AL3C
  RAM:      Winbond W9751G6KB-25 64 MiB DDR2-800
  Flash:    Winbond W25Q128FVSG 16 MiB
  WLAN:     Wi-Fi4 2.4 GHz 3*3 + 5 GHz 3*3
  LAN:      LAN ports *4
  WAN:      WAN port *1
  Buttons:  reset *1 + wps *1
  LEDs: ethernet *5, power, wlan, wps

MAC Address:
  use      address               source1          source2
  label    40:9b:xx:xx:xx:3c     lan && wlan      u-boot,env@ethaddr
  lan      40:9b:xx:xx:xx:3c     devdata@0x3f     $label
  wan      40:9b:xx:xx:xx:3f     devdata@0x8f     $label + 3
  wlan2g   40:9b:xx:xx:xx:3c     devdata@0x5b     $label
  wlan5g   40:9b:xx:xx:xx:3e     devdata@0x76     $label + 2

Install via Web UI:
  Apply factory image in the stock firmware's Web UI.

Install via Emergency Room Mode:
  DIR-859 A1 will enter recovery mode when the system fails to boot or
  press reset button for about 10 seconds.

  First, set computer IP to 192.168.0.5 and Gateway to 192.168.0.1.
  Then we can open http://192.168.0.1/ in the web browser to upload
  OpenWrt factory image or stock firmware. Some modern browsers may
  need to turn on compatibility mode.
```

### CODE REVIEW

File: target/linux/ath79/generic/config-default [hunk 1]
```diff
@@ -15,6 +15,8 @@ CONFIG_MARVELL_PHY=y
 CONFIG_MICREL_PHY=y
 CONFIG_MTD_REDBOOT_PARTS=y
 CONFIG_MTD_SPLIT_EVA_FW=y
+CONFIG_NVMEM_SYSFS=y
+CONFIG_NVMEM_U_BOOT_ENV=y
```
Comment (DragonBluep) [inline target/linux/ath79/generic/config-default:19]: @musashino205 I picked this nvmem commit from your original PR https://github.com/openwrt/openwrt/pull/11228 and add the `CONFIG_NVMEM_SYSFS=y`.

The entire feature increases the kernel size by about 1.5 KiB.

**Comment by hauke** [comment]
I merged #3997, could you please rebase your changes and also use target/linux/ath79/dts/qca9563_dlink_dir-8x9-a1.dtsi if possible.

**Comment by DragonBluep** [comment]
> I merged #3997, could you please rebase your changes and also use target/linux/ath79/dts/qca9563_dlink_dir-8x9-a1.dtsi if possible.

I have rebased and run test it. Everything works well.

File: target/linux/ath79/dts/qca9563_dlink_dir-859-a3.dts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+			color = <LED_COLOR_ID_GREEN>;
+			function = LED_FUNCTION_WPS;
+			gpios = <&gpio 8 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&ath10k {
+	nvmem-cells = <&calibration_ath10k>, <&macaddr_bdcfg_ethaddr 2>;
+	nvmem-cell-names = "calibration", "mac-address";
```
Comment (DragonBluep) [inline target/linux/ath79/dts/qca9563_dlink_dir-859-a3.dts:52]: I didn't remove these properties in the common dtsi but just overrode it here.

**Pull request closed without merging**